flask_security/views.py

```diff
@@ -267,33 +267,51 @@ def reset_password(token):
     """View function that handles a reset password request."""
 
     expired, invalid, user = reset_password_token_status(token)
+    form_class = _security.reset_password_form
+
+    if request.json:
+        form = form_class(MultiDict(request.json))
+    else:
+        form = form_class()
 
     if invalid:
         do_flash(*get_message('INVALID_RESET_PASSWORD_TOKEN'))
+        form.errors['invalid'] = get_message('INVALID_RESET_PASSWORD_TOKEN')
```

Comment:
Can you move the `form.errors['invalid']` above the `do_flash` and reuse the message?

```python
invalid_message = form.errors['invalid'] = get_message(
    'INVALID_RESET_PASSWORD_TOKEN'
)
do_flash(*invalid_message)
```